kolibri/auth/test/test_collections_and_roles.py

```diff
@@ -2,204 +2,143 @@
 
 from django.test import TestCase
 
-from kolibri.auth.models import FacilityUser, Facility, Classroom, LearnerGroup, FacilityAdmin, Coach, Learner, \
-    Role, Collection, HierarchyNode
+from kolibri.auth.models import FacilityUser, Facility, Classroom, LearnerGroup, Role, Collection, FacilityDataset
 
 
 class CollectionRemovalTestCase(TestCase):
     """
-    Tests that removing Roles and sub-Collections from a Collection does not create orhpans.
+    Tests that removing users from a Collection properly deletes the corresponding Role object, and that deleting
+    a Collection also deletes the sub-Collections underneath it.
     """
+
     def setUp(self):
+        self.dataset = FacilityDataset.objects.create()
         user1, user2, user3 = self.user1, self.user2, self.user3 = (
-            FacilityUser.objects.create(username='foo'),
-            FacilityUser.objects.create(username='bar'),
-            FacilityUser.objects.create(username='baz'),
+            FacilityUser.objects.create(username='foo', dataset=self.dataset),
+            FacilityUser.objects.create(username='bar', dataset=self.dataset),
+            FacilityUser.objects.create(username='baz', dataset=self.dataset),
         )
-        self.lg = LearnerGroup.objects.create()
-        self.lg.add_learner(user1)
 
-        self.cr = Classroom.objects.create()
+        self.f = Facility.objects.create(dataset=self.dataset)
+        self.f.add_admin(user3)
+
+        self.cr = Classroom.objects.create(dataset=self.dataset, parent=self.f)
         self.cr.add_coach(user2)
-        self.cr.add_learner_group(self.lg)
 
-        self.f = Facility.objects.create()
-        self.f.add_admin(user3)
-        self.f.add_classroom(self.cr)
+        self.lg = LearnerGroup.objects.create(dataset=self.dataset, parent=self.cr)
+        self.lg.add_learner(user1)
 
     def test_remove_learner(self):
-        self.assertEqual(Learner.objects.count(), 1)
+        self.assertEqual(Role.objects.filter(user=self.user1, kind=Role.KIND_LEARNER, collection=self.lg).count(), 1)
```

Comment:
There will be some nicer APIs for checking roles/relations once the permissions stuff is integrated. I will spec these interfaces out in the permissions tests (and perhaps in this file as well) as the first step of integrating that stuff.